upgrade/scripts/upgrade/prerequisites.sh

```diff
@@ -1160,6 +1160,46 @@ if [[ ${state_recorded} == "0" && $(hostname) == "${PRIMARY_NODE}" ]]; then
       csi handoff bss-update-param --set "${bootparameter}"
     done
 
+    # Get a list of NCNs.
+    if IFS=$'\n' read -rd '' -a NCN_XNAMES; then
+      :
+    fi <<< "$(cray hsm state components list --role Management --type Node --format json | jq -r '.Components | map(.ID) | join("\n")')"
+
+    # If no NCNs are found we should exit, otherwise if forces its way forward then NCNs will be missing critical packages.
+    if [ "${#NCN_XNAMES[@]}" -eq '0' ]; then
+      echo >&2 'No NCN xnames were found in HSM! Aborting.'
+      exit 1
+    fi
+
+    params=""
+    error=0
+
+    # Loop through one at a time. If `--hosts` isn't provided, we will error out on the 'Global' key.
+    for ncn_xname in "${NCN_XNAMES[@]}"; do
+      printf "% -15s: " "${ncn_xname}"
+
+      params=$(cray bss bootparameters list --hosts "${ncn_xname}" --format json | jq '.[] |."params"' \
+        | sed -E \
+          -e 's/ip=hsn[0-9]+:auto6//g' \
+          -e 's/ifname=hsn[0-9]+:[0-9a-fA-F:]{17}\s+//' \
+        | tr -d \")
```

Comment:
I'm curious what the "tr -d \" is doing.

Reply:
It's removing double quotes. We don't want double quotes at the beginning or end of our params. For the sake of quickness I copied what was in 1.5, but I just updated tr to use sed instead.

Reply:
OK that makes sense.   I was overthinking what that was doing.